src/AzureAuth/ExceptionListToStringConverter.cs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text.Json;
+    using Microsoft.Identity.Client;
+
+    /// <summary>
+    /// The exception list to string converter.
+    /// </summary>
+    public static class ExceptionListToStringConverter
+    {
+        /// <summary>
+        /// Executes the convertion of exceptions to a JSON string.
+        /// </summary>
+        /// <param name="exceptions">List of exceptions.</param>
+        /// <returns>JSON format of exception list.</returns>
+        public static string Execute(IEnumerable<Exception> exceptions)
+        {
+            List<SerializableException> serializableExceptions = new List<SerializableException>();
+            foreach (Exception exception in exceptions)
+            {
+                var customException = new SerializableException(exception);
+                serializableExceptions.Add(customException);
+            }
+
+            return JsonSerializer.Serialize(serializableExceptions);
+        }
+
+        /// <summary>
+        /// CustomException class with only select properties of Exception class.
+        /// </summary>
+        public class SerializableException
+        {
+            /// <summary>
+            /// Initializes a new instance of the <see cref="SerializableException"/> class.
+            /// </summary>
+            /// <param name="exception">Exception.</param>
+            public SerializableException(Exception exception)
+            {
+                if (exception == null)
+                {
+                    throw new ArgumentNullException(nameof(exception));
+                }
+
+                if (exception.Message != null)
+                {
+                    this.Message = exception.Message.Replace("\r", string.Empty);

```

Comment:
This data sanitization is only being done in the constructor, but this is public property that can be reset. As a general guideline, I would move this kind of cleaning into the `set` method of the property so that the logic is "owned" by the property itself. That way the property always has `x` removed no matter where it is set from.

Looking back it took

Reply:
A second comment on this, it took me a minute to remember why we are only removing `\r` here, I recalled it's because we want to only have \n in the messages to make them easier to parse. I can't remember why this was still desired after the move to use JSON serialization though. I'm not able to think of a reason why we would need to be able to always split the `Message` on newlines, in Kusto, since each Message is going to be in it's own object.

Reply:
Ah re-reading, I had said we don't need to strip them, but we can force them to always be LF, which is what this does. So only pay attention to my first note here about data consistency, cleaning in the setter :)